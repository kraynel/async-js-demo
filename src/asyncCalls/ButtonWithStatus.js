// @flow

import React, { PureComponent } from 'react';

type PropsType = {
  buttonText: string,
  toggleCall: () => void,
  strangerStatus: string
};

export default class ButtonWithStatus extends PureComponent<PropsType> {
  render() {
    return (
      <div className="buttonWithStatus">
        <button
          className="buttonToggleUpsideDown"
          onClick={this.props.toggleCall}
        >
          {this.props.buttonText}
        </button>
        <span>{this.props.strangerStatus}</span>
      </div>
    );
  }
}
