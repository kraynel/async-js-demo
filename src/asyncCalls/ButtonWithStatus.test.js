import React from 'react';
import renderer from 'react-test-renderer';

import ButtonWithStatus from './ButtonWithStatus';

describe('ButtonWithStatus', () => {
  it('should display a button with a status', () => {
    const toggleCall = jest.fn();
    const component = renderer.create(
      <ButtonWithStatus
        buttonText="Some text"
        strangerStatus="some status"
        toggleCall={toggleCall}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
