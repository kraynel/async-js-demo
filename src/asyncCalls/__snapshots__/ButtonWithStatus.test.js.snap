// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithStatus should display a button with a status 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Some text
  </button>
  <span>
    some status
  </span>
</div>
`;
